mypy/subtypes.py

```diff
@@ -143,10 +143,15 @@ def visit_instance(self, left: Instance) -> bool:
             rname = right.type.fullname()
             if not left.type.has_base(rname) and rname != 'builtins.object':
                 return False
-
             # Map left type to corresponding right instances.
             t = map_instance_to_supertype(left, right.type)
-
+            # Check that one of the types does not have type args or
+            # the number of type args is the same and
+            # each left type arg is acceptable in place of the matching right one
+            if not t.args or not right.args:
+                return True
+            if len(t.args) != len(right.args):
+                return False
```

Comment:
This assert fails 298 times on `mypy mypy`.

I checked just one example, it's basically something like `x = {1: 'a'}`. In this example, in order to verify that the `dict` constructor has the right argument, `is_subtype` is asked whether `Tuple[builtins.int, builtins.str]` is a subtype of `Tuple[builtins.int*, builtins.str*]` (what does the * on the RHS mean btw?).

That in turn requires verifying that the fallback of LHS (`builtins.tuple[builtins.object]`) is a subtype of the fallback of RHS (`builtins.tuple`). `visit_instance` is used to verify that.

`map_instance_to_supertype` returns right away here because both sides are `builtins.tuple`. Then the code we're discussing is reached, with `t.args` having length 1 (`[builtins.object]`) and `right.args` having length 0.

I thought that's not a problem, since 0 arguments is treated as `Any`; and I thought `Any` is also ok. But I thought in every other situation, it's not a bug, but merely the type args don't match (so I can return `False`).

But I may be wrong; are you saying that `map_instance_to_supertype` is supposed to do that check, so if I see mismatched type args after it returns, it's a bug?